snuba/web/query.py

```diff
@@ -61,7 +62,21 @@ def parse_and_run_query(
     """
     Runs a Snuba Query, then records the metadata about each split query that was run.
     """
-    # from_clause = request.query.get_from_clause()
+
+    entity = request.query.get_from_clause()
+    assert isinstance(entity, Entity)
```

Comment:
don't do this, it can break subqueries

Reply:
Hm, okay. Good to know. So what is the best way to get the actual table or entity that a query is hitting, at this stage in the pipeline? Or perhaps this check shouldn't live here?

Reply:
The from clause is not guaranteed to be an Entity.

```
from_clause = request.query.get_from_clause()
if isinstance(from_clause, Entity):
    # the rest of your logic
```